src/compiler/emitter.ts

```diff
@@ -570,9 +540,16 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 forEach(sourceFiles, emitSourceFile);
 
                 writeLine();
-                writeEmittedFiles(writer.getText(), jsFilePath, /*writeByteOrderMark*/ compilerOptions.emitBOM);
+
+                const sourceMappingURL = sourceMap.getSourceMappingURL();

```

Comment:
Right now, the SourceMapWriter's only job is to write the source map, while the emitter's only job is to write the JS file. I'd rather not further conflate the two and would rather keep the logic here.